client/cli/build.rs

```diff
@@ -0,0 +1,21 @@
+use std::{env, fs::File, process::Command};
+fn main() {
+    let prev_hash_var = env::var("GIT_HASH");
+    if let Ok(prev_hash) = prev_hash_var {
+        if prev_hash.len() < 4 || &prev_hash[0..4] != "fedi" {
```

Comment:
I don't understand this one. Users that want to override the hash should be able to just set any, no?

Possibly we should name the variable itself `FEDIMINT_GIT_HASH` if we are afraid of collisions with other software, etc.